riscv/src/register/satp.rs

```diff
@@ -14,27 +16,35 @@ impl Satp {
     }
 
     /// Current address-translation scheme
+    ///
+    /// **WARNING**: panics if the field has an invalid variant.
     #[inline]
     #[cfg(target_pointer_width = "32")]
     pub fn mode(&self) -> Mode {
-        match self.bits & (1 << 31) != 0 {
-            false => Mode::Bare,
-            true => Mode::Sv32,
-        }
+        self.try_mode().unwrap()
+    }
+
+    /// Attempts to get the current address-translation scheme.
+    #[inline]
+    #[cfg(target_pointer_width = "32")]
+    pub fn try_mode(&self) -> Result<Mode> {
+        ((self.bits >> 31) as u8).try_into()
     }
```

Comment:
Not sure if it is worth it to have the `try_mode` and `mode` methods. Theoretically, the struct should never hold invalid values, right? Maybe only the `mode` method with the `unwrap()` is enough.

Reply:
I wasn't sure, either. For consistency with the `target_pointer_width = "64"` implementation, I included it.

I don't think it really harms anything to have the `try_mode` function for the `target_pointer_width = "32"` implementation.

It would also allow for less feature-gating downstream to keep `try_mode` for both implementations.